### PR TITLE
feat: add api key CRUD methods

### PR DESCRIPTION
**Pull request by stakach**
blocked by placeos/models#106

### CODE REVIEW

File: src/placeos-rest-api/controllers/api_keys.cr [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+module PlaceOS::Api
+  class ApiKeys < Application
+    base "/api/engine/v2/api_keys/"
+
+    before_action :check_admin
+    before_action :body, only: [:create, :update, :update_alt]
+
+    getter current_api_key : Model::ApiKey { find_api_key }
+
+    def index
```
Comment (caspiano) [inline src/placeos-rest-api/controllers/api_keys.cr:12]: We’ll need to add the model into rubber-soul as well

File: spec/helper.cr [hunk 2]
```diff
@@ -46,13 +46,25 @@ def clear_tables
...
+                    PlaceOS::Model::ApiKey,
                   } %}
         Promise.defer { {{t.id}}.clear },
       {% end %}
     )
   {% end %}
 end
 
+# Yield an authenticated user, and a header with X-API-Key set
+def xapi_authentication(scope = ["public"] of String)
```
Comment (caspiano) [inline spec/helper.cr:57]:
```suggestion
def x_api_authentication(scope = ["public"] of String)
```
Just to be consistent with the method

**Review by caspiano — COMMENTED** [review]
We can also bump the dockerfile to 1.1.0 🚀

File: src/placeos-rest-api/controllers/api_keys.cr (hunk 1 above)
Comment (stakach) [reply src/placeos-rest-api/controllers/api_keys.cr:12]: should be in the next nightly

File: spec/helper.cr (hunk 2 above)
Comment (stakach) [reply spec/helper.cr:57]: kk will do

File: src/placeos-rest-api/controllers/api_keys.cr (hunk 1 above)
Comment (caspiano) [reply src/placeos-rest-api/controllers/api_keys.cr:12]: The class needs to be included here  https://github.com/PlaceOS/rubber-soul/blob/754cb9476ab67c74468104d24de80f7a3c5b9e07/src/config.cr#L9

**Review by caspiano — APPROVED** [review]
LGTM! could you squash this one? quite a few of the commit messages are non-standard

**Pull request merged**